### ISSUE TITLE
Design Improvement

### ISSUE THREAD

**Issue by indpurvesh** [mage2/laravel-ecommerce#11]
I am thinking to do a design improvement now for front end and backend both.

**Comment by kmg101** [comment]
Hey, I'm interested in helping. This is an awesome project and will like to see it grow.

**Comment by indpurvesh** [comment]
I am glade to hear it. I have added a themes/mage2/basic theme.  It does have assets folder and one home view file only for now (which override the view from modules/base/Mage2/Dashboard/views/home/index.blade.php). I hope this information make sense.

If you add another theme with bootstrap framework it would be great. thanks @kmg101

**Comment by kmg101** [comment]
Yes, I will take a look at it this week. Thanks

**Comment by indpurvesh** [comment]
Thanks @kmg101

**Comment by designviacode** [comment]
Hey i'm really interested in working on this project. Please let me know what can i pick up on .. like right now .. :)

**Comment by indpurvesh** [comment]
@designviacode  Very pleased to hear that you like to contribute to project. 
If you help it with creating first basic theme design.  Let me give you an example of how you can override the template. Let say if you want to override  product view template. Currently its loading view from: modules/base/Mage2/Catalog/views/product/view.blade.php you can copy this file from here to theme folder: themes/mage2/basic/product/view.blade.php

**Comment by zaisahil** [comment]
Getting an error: 

RuntimeException in Encrypter.php line 43:
The only supported ciphers are AES-128-CBC and AES-256-CBC with the correct key lengths.

Please help!

**Comment by Eggotron** [comment]
@zaisahil . Check that you hace your .env file properly configured. Procced to your command line and write : php artisan key:generate

Done.

**Comment by indpurvesh** [comment]
thanks @Eggotron

**Comment by sandsax** [comment]
I will study now this solution if works for me i will do a template and add to project ok?

Thanks

**Comment by ihabhamad** [comment]
i now working on front end theme and some enhancement

**Comment by krahamath** [comment]
I need to contribute front and backend. I have other projects in my local those are works with php5.xx but this works only with php 7xx so any idea to make this work at local?

**Comment by indpurvesh** [comment]
@krahamath 
If you have a PHP 7.XXX installed in local it wont stop PHP 5.** Projects.

**Comment by krahamath** [comment]
It affect so that I am asking

**Comment by nteej** [comment]
Hai,All Members,

Me too interested to help on this project.Also need to be a backer from next month end.How I can join?

**Comment by indpurvesh** [comment]
All the Contribution is always welcome. If you want to do some design improvements i have some ideas in my mind but its not priority for me at this stage:
 - Product Page add Zoom Image or lightbox for Product Images
 - Category Page: make all product card same height. so all row gets alignments.
- Cart and checkout Page I am using a table. I would like to replace it with Div may be?

**Comment by nteej** [comment]
Ok sure.1st of all I'll do develop these things .I can do implement one or 2 things today itself

**Comment by nteej** [comment]
I can fix these issue due I'm fixing them all for my deployment.

**Comment by indpurvesh** [comment]
You can use dev branch

**Comment by nteej** [comment]
I'm currently on it.But I cant commit my changes due unauthorize.You need to add me as a collaborator.Then only I can commit my code.

**Comment by nteej** [comment]
Thanks.Also do the same for framework repo.I've updated that too.

**Comment by nteej** [comment]
Also you may be like to this type UI that I've aready used my my own laravel app.
![screenshot from 2018-06-30 13-05-07](https://user-images.githubusercontent.com/24771749/42122909-cce81c20-7c66-11e8-8b26-b4511192e9a1.png)

![screenshot from 2018-06-30 13-09-42](https://user-images.githubusercontent.com/24771749/42122915-ed4c9220-7c66-11e8-9c10-650ef40f7620.png)

**Comment by nteej** [comment]
Also I'm happy to add Megamenu function if you willing to have.
![screenshot from 2018-06-30 13-11-10](https://user-images.githubusercontent.com/24771749/42122928-2159b408-7c67-11e8-89c4-089a2b85742e.png)

**Comment by indpurvesh** [comment]
As a demo or default I would prefer a simple design  if you want you can put your theme as a custom theme

**Comment by indpurvesh** [comment]
As you mentioned mega menu and others comes as a separate modules.

**Comment by nteej** [comment]
ok sure.Lets follow that structure.its quite easy.

**Comment by ghermans** [comment]
@indpurvesh  where are you storing the backend theme for avored? i will help to give it a modern look :)

**Comment by ghermans** [comment]
@indpurvesh the current method that is used to display categories isn't very useful if you have a lot of categories.
Using a megamenu could resolve the issue or we could make a left menu like suggested above by @nteej  's comment. https://github.com/avored/laravel-ecommerce/issues/11#issuecomment-401524471

**Comment by indpurvesh** [comment]
You are right. But there is an option to put a category into a submenu as well. I am hoping to make a mega menu as a custom module instead of build in feature. But again I will think about it. 

Instead I am thinking of adding a group menu into an admin let say: (main top menu, left menu or footer menu) then inside a code they just do `$menus = Menu::get('footer-menu')` I know this feature does not exist right now but I have it in my queue and once I hope then designer/developer will then use those function and put the menu as they like in their theme.

@ghermans What's your opinion?

**Comment by indpurvesh** [comment]
@ghermans Your front design looks very nice. Thanks for the hard work that you have put in.

**Comment by ghermans** [comment]
@indpurvesh i do understand that you want to keep the default template simple as possible, but keeping something simple does not mean that the  end-user / customer experience may suffer on it.    

What i have seen on other e-commerce platforms like OpenCart & Magento is that most of the time end-users and even designers use the default theme to add / modify there own style to it,
so i would suggest to consider adding more features that benefit the end-user / customer experience by default.    

Also thank you for your feedback regarding the front-end improvements, i'm not done with it yet but i must admit that it looks better then before :) also if you have any suggestions regarding the design ping me and i will try to help

**Comment by indpurvesh** [comment]
@ghermansI agree that nowadays thing has changed a lot. Letus make a mega menu.

**Comment by indpurvesh** [comment]
@ghermans Lets Add most of my todo list into here. 

https://github.com/orgs/avored/projects/1

**Comment by ghermans** [comment]
The link isn't working

**Comment by ghermans** [comment]
@indpurvesh could you maybe add the current open issues also to the project list?

**Comment by indpurvesh** [comment]
I have added most of them.

**Comment by ghermans** [comment]
@indpurvesh i'm busy with changing the datagrid for the backend theme to default html tables instead divs.
![datagridknipsel](https://user-images.githubusercontent.com/1594411/44644430-5e984b00-a9d4-11e8-93d7-82b5c5ffda53.PNG)

**Comment by indpurvesh** [comment]
@ghermans no worries friend.

Possibly yes you can use `tables`

**Comment by ihabhamad** [comment]
@indpurvesh you can use my mega menu laravel package 
https://github.com/ihabhamad/LaravelMegaMenu

**Comment by ihabhamad** [comment]
New version in progress and well be coming soon but iam very busy now

**Comment by ghermans** [comment]
@indpurvesh i am still busy with upgrading the back-end theme but i expect to push the changes tomorrow or during the weekend.

![backend-dash](https://user-images.githubusercontent.com/1594411/45844869-e3e8f400-bd23-11e8-915d-4f0cc2ef4513.PNG)

![backend-dash2](https://user-images.githubusercontent.com/1594411/45844868-e3e8f400-bd23-11e8-9456-cfbbc05bd05e.PNG)

**Comment by indpurvesh** [comment]
@nteej or @designviacode anyone have some idea about improving the frontend design?
